src/main/scala/com/redhat/et/silex/util/optionalArg.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * This file is part of the "silex" library of helpers for Apache Spark.
+ *
+ * Copyright (c) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.c
+ */
+
+package com.redhat.et.silex.util
+
+/**
+ * Provides idiomatic optional parameter values on top of [[Option]], but which can be passed in as
+ * raw values, instead of having to pass in Some(value)
+ * {{{
+ * import com.redhat.et.silex.util.OptionalArg
+ *
+ * def possiblyFilter(data: Seq[Int], filterMax: OptionalArg[Int] = None) = {
+ *   // Treat an OptionalArg like an Option, for idiomatic handling of unset vals as None
+ *   if (filterMax.isEmpty) data else data.filter(_ <= filterMax.get)
+ * }
+ *
+ * val data = Seq(1, -1, 2)
+ *
+ * // default filterMax is None
+ * possiblyFilter(data) // returns data unfiltered
+ *
+ * // provide argument as (-1), instead of having to specify Some(-1)
+ * possiblyFilter(data, filterMax = -1) // returns Seq(-1)
+ *
+ * // Support full Option methods and implicit conversion to Option
+ * import OptionalArg.fullOptionSupport
+ * OptionalArg("xxx").map(_ + "zzz").get // returns "xxxzzz"
+ * }}}
+ *
+ * @tparam A The underlying value type.
+ * @param option The raw Option value.  May be implicitly constructed from raw value of type A
+ */
+class OptionalArg[+A](val option: Option[A]) extends AnyVal {
```

Comment:
Not sure how big a deal it is, but it does allow a parameter value that is a sub-type to be passed.
I think if it was invariant, that would fail, but I haven't experimented.